app/src/analytics/analytics-client.ts

```diff
@@ -1,91 +1,168 @@
-    LOCAL = 'local',
-    DEV = 'dev',
-    STAGING = 'staging',
-    PROD = 'prod'
+interface EventPayload {
+	action: string,
+	actionSubject: string,
+	attributes?: AnalyticsAttributes
 }
 
+interface Event {
+	userId?: string;
+	anonymousId: string;
+	event: string;
+	properties: {
+		action: string;
+		actionSubject: string;
+		attributes: Record<string, string>;
+		env: string;
+		eventType: string;
+		origin: string;
+		product: string;
+		tenantId: string;
+		tenantIdType: string;
+		source: string;
+		userIdType?: string;
+	};
+	type: string;
+	timestamp: string;
+	_metadata: Metadata;
+	context: Context;
+	messageId: string;
+  }
+
+interface Payload {
+	batch: Event[];
+	timestamp: string;
+	sentAt: string;
+  }
+
+interface Options {
+	method: string;
+	body: string;
+	headers: {
+		'Content-Type': string;
+	};
+  }
+
+  const environments = {
+	LOCAL: 'local',
+	DEV: 'dev',
+	STAGING: 'stg',
+	PROD: 'prod'
+  };
+
+  const TRACK_EVENT_TYPE = 'track';
+  const PRODUCT = 'jenkinsForJira';
+  const ORIGIN = 'server';
+  const TENANT_ID_TYPE = 'cloudId';
+
 // eslint-disable-next-line max-len
 export const sendAnalytics = async (cloudId: string, eventPayload: EventPayload, accountId?: string, anonymousId?: string): Promise<void> => {
-    sendEvent(cloudId, eventPayload, accountId, anonymousId)
-        .then(() => console.log('Analytics event processed'))
-        .catch((e) => console.error({ e }, 'Failed to send analytics event'));
+	console.info('Analytics Request');
+
+	sendEvent(cloudId, eventPayload, accountId, anonymousId)
+		.then(() => console.info('Analytics event processed'))
+		.catch((e) => console.error({ e }, 'Failed to send analytics event'));
 };
 
-// eslint-disable-next-line max-len
+const getAnalyticsEnvironmentUrl = () => {
+	if (isProductionEnv()) {
+        return process.env.ANALYTICS_URL;
+	}
+	return process.env.ANALYTICS_STAGE_URL;
+};
+
+// eslint-disable-next-line max-len,@typescript-eslint/no-unused-vars
 const sendEvent = async (cloudId: string, eventPayload: EventPayload, accountId?: string, anonymousId?: string): Promise<void> => {
-    const analyticsClient = await getAnalyticsClient();
-    const {
-        eventName, attributes, actionSubject, action
-    } = eventPayload;
-
-    if (!analyticsClient || !isProductionEnv()) {
-        console.warn('Analytics sendEvent skipped: @atlassiansox/analytics-node-client module not found or environment not production.');
-        return;
-    }
-
-    const accountDetails = getAccountDetails(accountId, anonymousId);
-
-    await analyticsClient.sendTrackEvent({
-        ...accountDetails,
-        tenantIdType: 'cloudId',
-        tenantId: cloudId,
-        trackEvent: {
-            source: eventName,
-            action: action || 'unknown',
-            actionSubject: actionSubject || eventName,
-                attributes: {
-                    ...attributes
-                }
-        }
-    });
+	const url = getAnalyticsEnvironmentUrl();
+	if (!url) {
```

Comment:
Is there a situation where we would not have the URL? It looks like we are setting it just above and it will always default to `ANALYTICS_STAGE_URL` in the event that something goes wrong in `isProductionEnv()`.

Reply:
I'm assuming you are just "returning" rather than erroring out because analytics shouldn't block the action that is occurring before calling analytics e.g. You wouldn't want to stop a server from connecting because analytics failed.

Reply:
very common for local dev if you haven't set up you variables. and yeah, good to know if its not there but dont interfere